test/test_autograd.py

```diff
@@ -4185,6 +4185,31 @@ def test_power_function(self):
         c.backward()
         self.assertEqual(b.grad, torch.tensor([-inf, 0., 0.]))
 
+    def test_nansum_with_nans(self):
```

Comment:
@albanD to check the derivative for these nan* methods and look at this test.

Reply:
This looks good.
It would be nice to be able to add this directly in the common method invocation. But we can't really do it atm.